### PR TITLE
Update rule S5122 [python/php/csharp] taint/user-controlled code examples

### PR DESCRIPTION
**Pull request by eric-therond-sonarsource**

### CODE REVIEW

File: rules/S5122/csharp/rule.adoc [hunk 1]
```diff
@@ -66,9 +66,17 @@ public HttpResponseMessage Get() => new HttpResponseMessage()
 };
 ----
 
+User-controlled origin
+
+----
+String origin = Request.Headers["Origin"];
+Response.Headers.Add("Access-Control-Allow-Origin", origin);
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S5122/csharp/rule.adoc:73]: Missing "sensitive" comment.

File: rules/S5122/csharp/rule.adoc [hunk 2]
```diff
@@ -124,6 +131,18 @@ public HttpResponseMessage Get() => new HttpResponseMessage()
 };
 ----
 
+
+User-controlled origin validated with an allow-list
+
+----
+String origin = Request.Headers["Origin"];
+
+if(TRUSTED_ORIGINS.Contains(origin))
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S5122/csharp/rule.adoc:140]: Very minor: in the code samples of C# there usually seems to be a space after the `if`.

Comment (hendrik-buchwald-sonarsource) [reply rules/S5122/csharp/rule.adoc:140]: Also very minor: is it common to write constants upper-case in C# like in some other languages?

File: rules/S5122/php/rule.adoc [hunk 3]
```diff
@@ -49,9 +64,23 @@ use Symfony\Component\HttpFoundation\Response;
...
+User-controlled origin validated with an allow-list
+
+----
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Request;
+
+$origin = $request->headers->get('Origin');
 
-$response->headers->set('Access-Control-Allow-Origin',$trusteddomain); // Compliant
+if(in_array($origin, $TRUSTED_ORIGINS)) {
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S5122/php/rule.adoc:81]: Very minor: in the code samples of PHP there usually seems to be a space after the if.

File: rules/S5122/python/rule.adoc [hunk 4]
```diff
@@ -22,6 +22,14 @@ app = Flask(__name__)
 CORS(app, resources={r"/*": {"origins": "*", "send_wildcard": "True"}}) # Sensitive
 ----
 
+User-controlled origin:
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S5122/python/rule.adoc:25]: For Python the sentences in the code samples end with a colon, for the languages not. Personally, I prefer it with colons.

**Review by hendrik-buchwald-sonarsource — APPROVED** [review]
I left comments for some minor issues.

File: rules/S5122/csharp/rule.adoc (hunk 2 above)
Comment (eric-therond-sonarsource) [reply rules/S5122/csharp/rule.adoc:140]: I think the usual convention for C# is indeed camelCase for local variables
I will update that

File: rules/S5122/python/rule.adoc (hunk 4 above)
Comment (eric-therond-sonarsource) [reply rules/S5122/python/rule.adoc:25]: I noticed that and I prefer with colons too

File: rules/S5122/csharp/rule.adoc [hunk 5]
```diff
@@ -128,6 +135,18 @@ public HttpResponseMessage Get() => new HttpResponseMessage()
 };
 ----
 
+
+User-controlled origin validated with an allow-list:
+
```
Comment (loris-s-sonarsource) [inline rules/S5122/csharp/rule.adoc:140]:
```suggestion

[source,csharp]
```

File: rules/S5122/php/rule.adoc [hunk 6]
```diff
@@ -52,9 +67,23 @@ use Symfony\Component\HttpFoundation\Response;
...
     Response::HTTP_OK,
-    ['Access-Control-Allow-Origin' => $trusteddomain] /// Compliant
+    ['Access-Control-Allow-Origin' => $trusteddomain]
+);
+
+$response->headers->set('Access-Control-Allow-Origin', $trusteddomain); 
+----
+
+User-controlled origin validated with an allow-list:
+
```
Comment (loris-s-sonarsource) [inline rules/S5122/php/rule.adoc:77]:
```suggestion

[source,php]
```

File: rules/S5122/python/rule.adoc [hunk 7]
```diff
@@ -22,16 +22,24 @@ app = Flask(__name__)
...
+
```
Comment (loris-s-sonarsource) [inline rules/S5122/python/rule.adoc:26]:
```suggestion

[source,python]
```

File: rules/S5122/python/rule.adoc [hunk 8]
```diff
@@ -42,6 +50,15 @@ app = Flask(__name__)
 CORS(app, resources={r"/*": {"origins": "*", "send_wildcard": "False"}}) # Compliant
 ----
 
+User-controlled origin validated with an allow-list:
+
```
Comment (loris-s-sonarsource) [inline rules/S5122/python/rule.adoc:54]:
```suggestion

[source,python]
```

**Pull request merged**